README.md

```diff
@@ -142,7 +143,10 @@ example-gateway                 # root directory
 │   ├── clients                 # idl directory for client thrift files
 │   └── endpoints               # idl directory for endpoint thrift files
 ├── middlewares                 # config directory for modules of middleware module class
-│   └── transform-response      # config directory for a middleware named 'transform-response'
+│   ├── transform-response      # config directory for a middleware named 'transform-response'
+│   ├── mandatory               # directory for all mandatory middlewares
+│   │   └── log-publisher       # config directory for a mandatory middleware named 'log-publisher'
+│   └── mandatory.yaml          # config file describing mandatory middlewares and their execution order   
```

Comment:
Abhishek has brought to my attention that the contents of the `mandatory.yaml` file can actually be put inside the config field of `services/example-gateway/service-config.yaml`. For example, right now the `mandatory.yaml` file looks like this:
```
http:
  - mandatory_example2
  - mandatory_example
tchannel:
  - mandatory_example_tchannel
``` 
We can move this into `service-config.yaml` and that will look like this:
```
config: {
  mandatory_http_middleware:
  - mandatory_example2
  - mandatory_example
  mandatory_tchannel_middleware:
  - mandatory_example_tchannel
}
dependencies:
  endpoint:
  - bar
  - baz
  - contacts
  - googlenow
  - bazTChannel
  - multi
  - panic
  - panicTChannel
name: example-gateway
type: gateway
```
This makes sense because the mandatory middleware ordering is a service level configuration. Right now, I am leaning towards putting the middleware ordering in the `service-config.yaml` file. The code changes to implement this would be minimal (just changing where the config is read from). Any thoughts on this? @ChuntaoLu

Reply:
The downside of this approach is that we lose the genericness of "default dependencies", it becomes a special case for middlewares/services.  Think about the case that we might want endpoints to have default clients.

Reply:
I think "default dependencies" can still remain generic because the `mandatory.yaml` file is **not** necessary for "default dependencies".

"Default dependencies" was introduced so that anything in the `DefaultDepDirs` field will be included in the module's `module/dependencies.go` file. The standard "default dependencies" feature does **not** need to know about the ordering of the individual dependencies.

`mandatory.yaml` is only required to add the middlewares to the endpoints' execution stack in a specific order. The `mandatory.yaml` "feature" is built **on top** of "default dependencies". 

We can still have endpoints that have default clients, since those do not need a `mandatory.yaml` file.

Let me know if this isn't clear and we can discuss more

Reply:
Well said, makes sense.

Reply:
On second thought, this might not be a good idea. It's possible for to have more than one `service-config.yaml` file if there's more than one service in the `./services` folder. `example-gateway` has `echo-gateway` and `example-gateway` in its `./services` folder. If both of these services depend on the same endpoint, the ordering of the default middleware for this endpoint would be ambiguous since the list must come from either `example-gateway` or `echo-gateway`.

I'll keep the list as it is in a `default.yaml` file.

Reply:
Yes, multiple services are indeed a norm.